src/main.rs

```diff
@@ -240,6 +245,34 @@ fn get_languages_from_env() -> Vec<String> {
     )
 }
 
+fn spawn_editor(custom_pages_dir: &Path, file_name: &str) -> anyhow::Result<()> {
+    let res = create_dir_all(custom_pages_dir);
+    if !custom_pages_dir.is_dir() {
+        if res.is_err() {
+            res.context(format!("Fail to create dir {:?}", custom_pages_dir))?;
+        } else {
+            return Err(anyhow!("Fail to create dir {:?}", custom_pages_dir));
+        }
+    }
```

Comment:
@lengyijun Can you please explain why you are avoiding to just do `create_dir_all(...).context(...)?`? I feel like we are going on and on about this change and as long as we don't have clear evidence for why we _shouldn't_ use the simple version, I am not willing to merge anything more complicated. Your initial argument was that `create_dir_all` may return `Err` if the directory already exists, but we added a test that checks that this is not the case. What is there left to discuss?